pkg/apis/conditions.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apis
+
+import (
+	"context"
+
+	"knative.dev/pkg/apis"
+)
+
+// CopyConditions copies over explicitly marked conditions from one object to another
+func CopyConditions(ctx context.Context, from, to apis.ConditionsAccessor, allowed ...apis.ConditionType) {
```

Comment:
Couple nits:
1. This isn't specific to Serving, so we should probably put it into pkg
2. "Copy" feels like a misnomer because it doesn't convey the filtering.
3. Can we change `allowed` to a `func(apis.ConditionType) bool`, which would allow us to pass something that internally indexes things and uses `sets.String`

Reply:
Also, perhaps this is something where a conditionset can expose a compatible function?

Reply:
Moving discussion to https://github.com/knative/pkg/pull/1003